### ISSUE TITLE
File left in HOME directory

### ISSUE THREAD

**Issue by ndias** [gpodder/gpodder#1360]
Since version 3.11 some files are left in HOME directory, and a 0 file size is on the directory of the subscribed feed

An example 

-rw-r--r-- 1 ndias users 11809605 Aug  6 19:03 'BOLO DE 10 ANOS - #Short #Porta10Anos [jiKF-ndkdGs].mp4'
-rw-r--r-- 1 ndias users        0 Aug  7 08:46 'gPodder/Downloads/Porta dos Fundos/BOLO DE 10 ANOS - #Short #Porta10Anos.mp4'

I'm using version  gpodder-3.11.0-1.fc36.noarch from Fedora.

**Comment by auouymous** [comment]
What version of yt-dlp are you using?

**Comment by auouymous** [comment]
Are there any errors or warnings in the log file related to it? Gpodder tells yt-dlp what to name the file, which includes the directory path. Your yt-dlp is clearly ignoring that value and using its default name in $HOME, which includes the ID. There was an issue like this (and many others) before gpodder supported yt-dlp, but it has been pretty stable since then.

**Comment by ndias** [comment]
Yes, there are a warning in the Logs
2022-08-07 08:46:25,945 [tagging] WARNING: Unable to add tags to file '/home/ndias/gPodder/Downloads/Porta dos Fundos/BOLO DE 10 ANOS - #Short #Porta10Anos.mp4'

**Comment by auouymous** [comment]
That is post-download because the file it tried to tag was not a video (empty file). Was that the only warning?

**Comment by tpikonen** [comment]
> Gpodder tells yt-dlp what to name the file, which includes the directory path. Your yt-dlp is clearly ignoring that value and using its default name in $HOME, which includes the ID. 

The output file handling in yt-dlp is pretty much bonkers and attempts at setting it will break video post processing (see [here](https://github.com/yt-dlp/yt-dlp/issues/1844)).

I had to rewrite output file renaming when enabling post processing in #1363. The files left to $HOME are fixed by [setting](https://github.com/gpodder/gpodder/pull/1363/commits/e440e2109e11db6cfeb301d6521aad8a55da2ca7) the corrent 'paths' option to yt-dlp.

**Comment by ndias** [comment]
> That is post-download because the file it tried to tag was not a video (empty file). Was that the only warning?

Yes, is the only Warning related with this download, there are other warnings like this for every file left in $HOME directory.

**Comment by Strubbl** [comment]
I can confirm this issue with gpodder 3.11.0 in Archlinux. I have both, yt-dlp 2022.08.19 and youtube-dl 2021.12.17 installed. I do not know which one of the both was chosen to download episodes.

**Comment by auouymous** [comment]
@Strubbl It should be using yt-dlp (first choice), but I just pushed 2c0e70ca5fbfee082839d9fc080eb80cde3a66d3 to log the one it finds. Does it only place the file in $HOME, or does it also put the video ID in brackets as "title [video ID].mp4"?

**Comment by auouymous** [comment]
@tpikonen Setting the path by itself wouldn't fix this issue since yt-dlp is completely ignoring outtmpl. It would keep the files inside the podcast folder and gpodder would complain about unknown files at startup, for anyone who can see console or logs. However, I got `WARNING: --paths is ignored since an absolute path is given in output template`, is this a new warning or do your patches modify tempname?

```patch
@@ -301,6 +301,7 @@ class gPodderYoutubeDL(download.CustomDownloader):
 
     def fetch_video(self, url, tempname, reporthook):
         opts = {
+            'paths': {'home': os.path.dirname(tempname)},
             'outtmpl': tempname,
             'nopart': True,  # don't append .part (already .partial)
             'retries': 3,  # retry a few times
```

**Comment by tpikonen** [comment]
At least #1363 does not leave files to the podcast folder or print warning here. You also need to have the second to last patch in the stack in #1363 in order to rename the yt-dlp output to what gPodder expects.

**Comment by Strubbl** [comment]
> Does it only place the file in $HOME, or does it also put the video ID in brackets as "title [video ID].mp4"?

There is also the video ID in brackets.

**Comment by auouymous** [comment]
@tpikonen Do you feel that part of the PR is stable enough to move to a separate non-draft PR for testing? I'd like to get all of these issues fixed and a new release out in a week or two.

**Comment by Strubbl** [comment]
Excuse me, when is the next release planned, which has this bug fix?

**Comment by auouymous** [comment]
We are trying to build a new bundle which should fix the Mac issue and a release will happen shortly after that.

However, I haven't investigated yet, but I think #1382 (the fix for this issue) might cause an issue resuming downloads at startup because gpodder looks for *.partial, and yt-dlp now creates *.partial.{ext} files. This technically existed before #1382 as yt-dlp/youtube-dl would sometimes create files like this, but now it always does it.

I might also fix #1446, if possible.
